src/server.cpp

```diff
@@ -709,33 +709,31 @@ Response InternalServer::handle_search(const RequestContext& request)
   try {
     start = request.get_argument<unsigned int>("start");
   } catch (const std::exception&) {}
-  auto end = 25;
-  try {
-    end = request.get_argument<unsigned int>("end");
-  } catch (const std::exception&) {}
-  if (start>end) {
-    auto tmp = start;
-    start = end;
-    end = tmp;
-  }
-  if (end > start + MAX_SEARCH_LEN) {
-    end = start + MAX_SEARCH_LEN;
+  auto pageLength=25;
+  try{
+    pageLength=request.get_argument<unsigned int>("pageLength");
+  }catch(const std::exception&){}
+  if (pageLength > MAX_SEARCH_LEN) {
+    pageLength = MAX_SEARCH_LEN;
   }
+  if(pageLength==0)pageLength=25; // pageLength can't be zero
+  start=(start/pageLength)*pageLength; // changed start point according to page no.
```

Comment:
@veloman-yunkan no anything bad won't happen but I just thought it would be better if result start is the first result of the page before entering pagination as earlier it was a bit tough to understand setting range but it would be better now.

Reply:
@mgautierfr so which of these lines should i remove

Reply:
so I will remove these two then